core/src/test/java/org/apache/iceberg/rest/TestRESTCatalog.java

```diff
@@ -984,6 +969,73 @@ public void testTableSnapshotLoadingWithDivergedBranches(String formatVersion) {
         .hasSizeGreaterThan(Lists.newArrayList(table.snapshots()).size());
   }
 
+  @Test
+  public void lazySnapshotLoadingWithDivergedHistory() {
```

Comment:
this is porting the test I had in https://github.com/apache/iceberg/pull/8096, which reproduces the original issue